inst/htmljs/animint.js

```diff
@@ -543,56 +531,87 @@ var animint = function (to_select, json_file) {
           });
       }
       
-      // function to draw the grid lines when supplied with one of the elements
+      // function to draw major/minor grid lines 
       var grid_line = function(grid_background, grid_class) {
+        // if grid lines are defined
         if(Object.keys(grid_background).length > 1) {
           var col = grid_background.colour;
           var lt = grid_background.linetype;
           var size = grid_background.size;
           var cap = grid_background.lineend;
+        // group for grid lines
+        var grid = background.append("g")
+          .attr("class", grid_class);
 
+          // group for horizontal grid lines
+          var grid_hor = grid.append("g")
+            .attr("class", "hor");
           // draw horizontal grid lines if they are defined
           if(typeof grid_background.loc.y != "undefined") {
-            var draw_hor_line = function(element) {
-              svg.insert("line", "#plottitle")
-                .attr("x1", plotdim.xstart)
-                .attr("x2", plotdim.xend)
-                .attr("y1", function() { return scales[panel_i].y(element); })
-                .attr("y2", function() { return scales[panel_i].y(element); })
-                .attr("class", function() { return "grid " + grid_class; })
-                .style("stroke", col)
-                .style("stroke-linecap", cap)
-                .style("stroke-width", size)
-                .style("stroke-dasharray", function() {
-                  return linetypesize2dasharray(lt, size);
-                });
-            }
-            grid_background.loc.y.forEach(draw_hor_line);
+            // coercing y lines to array if necessary
+            if(typeof grid_background.loc.y == "number") grid_background.loc.y = [grid_background.loc.y];
```

Comment:
If it's a number, coerce it to an array.  If somebody knows of a better way to do this, let me know.